component/src/test/java/org/wso2/extension/siddhi/io/http/sink/HttpOAuthFailTestCase.java

```diff
@@ -0,0 +1,445 @@
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class HttpOAuthFailTestCase {
+    private static final Logger log = Logger.getLogger(HttpAuthTestCase.class);
+    public static final String CARBON_HOME = "carbon.home";
+    private AtomicInteger eventCount = new AtomicInteger(0);
+
+    /**
+     *Set carbon-home
+     */
+    private void setCarbonHome() {
+        Path carbonHome = Paths.get("");
+        carbonHome = Paths.get(carbonHome.toString(), "src", "test");
+        System.setProperty(CARBON_HOME, carbonHome.toString());
+        log.info("Carbon Home Absolute path set to: " + carbonHome.toAbsolutePath());
+    }
+
+    /**
+     * Creating test for publishing events with valid access token.
+     *
+     * @throws Exception Interrupted exception
+     */
+    @Test
+    public void testHTTPTextOAuthTrue() throws Exception {
+        setCarbonHome();
+        log.info("Creating test for publishing events with valid access token.");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.setExtension("xml-output-mapper", XMLSinkMapper.class);
+        String inStreamDefinition = "Define stream FooStream (message String,method String,headers String);" +
+                "@sink(type='http',publisher.url='https://localhost:8015/abc',method='{{method}}'" +
+                ",headers='{{headers}}',consumer.key='addConsumerKey', consumer.secret='addConsumerSecret', " +
+                "token.url='https://localhost:8005/token', @map(type='xml', @payload('{{message}}'))) "
+                + "Define stream BarStream (message String,method String,headers String);";
+        String query = (
+                "@info(name = 'query') "
+                        + "from FooStream "
+                        + "select message,method,headers "
+                        + "insert into BarStream;"
+        );
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+        InputHandler fooStream = siddhiAppRuntime.getInputHandler("FooStream");
+        siddhiAppRuntime.start();
+        HttpOAuthListenerHandler lst = new HttpOAuthListenerHandler(8005, 8015);
+        lst.run();
+        String payload = "<events>"
+                + "<event>"
+                + "<symbol>WSO2</symbol>"
+                + "<price>55.645</price>"
+                + "<volume>100</volume>"
+                + "</event>"
+                + "</events>";
+        fooStream.send(new Object[]{payload, "POST", "'Name:John','Age:23','Country:USA'," +
+                "'Authorization: Bearer yyyyy'"});
+        while (!lst.getHttpOAuthListener().isMessageArrive()) {
+            Thread.sleep(10);
```

Comment:
Could you confirm that this would not hang forever in case the response did not come?